File: src/types/transcode.ts
export interface TranscodeStatus {
  id: string
  status: TranscodeState
  progress: number
  outputPath?: string
  error?: string
}

export enum TranscodeState {
  Pending = 'Pending',
  Processing = 'Processing',
  Completed = 'Completed',
  Error = 'Error',
}

File: src-tauri/src/models/transcode.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct TranscodeStatus {
    pub id: String,
    pub status: TranscodeState,
    pub progress: f64,
    pub output_path: Option<String>,
    pub error: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum TranscodeState {
    Pending,
    Processing,
    Completed,
    Error,
}
